dp/cloud/python/magma/test_runner/tests/api_data_builder.py

```diff
@@ -0,0 +1,168 @@
+            antenna_gain: int = 15,
+            indoor_deployment: bool = True,
+            height_m: float = 12.5,
+            height_type: str = "agl",
+    ) -> CbsdAPIDataBuilder:
+        self.payload["installation_param"] = {
+            "latitude_deg": latitude_deg,
+            "longitude_deg": longitude_deg,
+            "antenna_gain": antenna_gain,
+            "indoor_deployment": indoor_deployment,
+            "height_m": height_m,
+            "height_type": height_type,
+        }
+        return self
+
+    def with_frequency_preferences(
+            self,
+            bandwidth_mhz: int = 20,
+            frequencies_mhz: List[int] = None,
+    ) -> CbsdAPIDataBuilder:
+        self.payload["frequency_preferences"] = {
+            "bandwidth_mhz": bandwidth_mhz,
+            "frequencies_mhz": frequencies_mhz or [],
+        }
+        return self
+
+    def with_capabilities(self, max_power=20, min_power=0, number_of_antennas=2):
+        self.payload['capabilities'] = {
+            'max_power': max_power,
+            'min_power': min_power,
+            'number_of_antennas': number_of_antennas,
+        }
+        return self
+
+    def with_desired_state(self, desired_state: str = "registered") -> CbsdAPIDataBuilder:
+        self.payload["desired_state"] = desired_state
+        return self
+
+    def with_expected_grant(
+        self, bandwidth_mhz: int = 10, frequency_mhz: int = 3625, max_eirp: int = 28,
+        grant_state="authorized",
+    ) -> CbsdAPIDataBuilder:
+        self.bandwidth_mhz = bandwidth_mhz
+        self.frequency_mhz = frequency_mhz
+        self.max_eirp = max_eirp
+        self.grant_state = grant_state
+        return self
+
+    def with_grant(
+        self, bandwidth_mhz: int = None, frequency_mhz: int = None, max_eirp: int = None, grant_state=None,
+    ) -> CbsdAPIDataBuilder:
+        self.payload['grant'] = {
+            'bandwidth_mhz': bandwidth_mhz or self.bandwidth_mhz,
+            'frequency_mhz': frequency_mhz or self.frequency_mhz,
+            'max_eirp': max_eirp or self.max_eirp,
+            'state': grant_state or self.grant_state,
+        }
+        return self
+
+    def with_max_eirp(self, max_eirp: int = 28) -> CbsdAPIDataBuilder:
+        self.payload['max_eirp'] = max_eirp
+        return self
+
+    def with_state(self, state: str = UNREGISTERED) -> CbsdAPIDataBuilder:
+        self.payload['state'] = state
+        return self
+
+    def with_cbsd_id(self, cbsd_id: str) -> CbsdAPIDataBuilder:
+        self.payload['cbsd_id'] = cbsd_id
+        return self
+
+    def with_is_active(self, is_active: bool) -> CbsdAPIDataBuilder:
+        self.payload['is_active'] = is_active
+        return self
+
+    def with_single_step_enabled(self, enabled: bool) -> CbsdAPIDataBuilder:
+        self.payload['single_step_enabled'] = enabled
+        return self
+
+    def build_grant_state_data(self, frequenzy_mhz=None, bandwidth_mhz=None, max_eirp=None) -> CBSDStateResult:
```

Comment:
What happened to the other build methods (this one is only for enodebd)? Do we no longer check GET requests?